src/normalize/endpoint.js

```diff
@@ -0,0 +1,9 @@
+'use strict'
+const UrlValueParser = require('url-value-parser')
+
+const normalizeEndpoint = function (endpointPath, extraMasks = [], placeholder = '#value') {
```

Comment:
Nitpick: I suggest you use something like `:id` instead of `#value`, because it matches more closely to the way routes are declared.